dcgmlib/src/DcgmClientHandler.cpp

```diff
@@ -229,8 +229,8 @@ dcgmReturn_t DcgmClientHandler::GetConnHandleForHostEngine(const char *identifie
     unsigned int portNumber = 0;
     char *p                 = NULL;
 
-    if (!addressIsUnixSocket)
-        p = strchr(&identifierTemp[0], ':');
+    if (!addressIsUnixSocket && identifierTemp[identifierTemp.size()-2] != ']')
```

Comment:
Could there be a possible out-of-bounds read here?
identifierTemp.size()-2 to make sense, I think identifierTemp.size() needs to be at least 2 here:
```suggestion
    if (!addressIsUnixSocket && identiferTemp.size() >= 2 && identifierTemp[identifierTemp.size()-2] != ']')
```